File: JUNE-2025/30-06-2025/p594.scala
object Solution {
  def findLHS(nums: Array[Int]): Int = {
    val freq = nums.groupBy(identity).view.mapValues(_.length).toMap
    var ans = 0

    for ((key, value) <- freq) {
      if (freq.contains(key + 1)) {
        ans = math.max(ans, value + freq(key + 1))
      }
    }
    ans
  }
}

File: JUNE-2025/30-06-2025/p594.go
import "math"

func findLHS(nums []int) int {
    freq := map[int]int{}
    for _, n := range nums {
        freq[n]++
    }

    ans := 0
    for key := range freq {
        if val, ok := freq[key+1]; ok {
            ans = int(math.Max(float64(ans), float64(freq[key]+val)))
        }
    }

    return ans
}